cmd/krew/cmd/index.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/krew/internal/gitutil"
+	"sigs.k8s.io/krew/pkg/constants"
+)
+
+// indexCmd represents the index command
+var indexCmd = &cobra.Command{
+	Use:    "index",
+	Short:  "Perform krew index commands",
+	Long:   "Perform krew index commands such as adding and removing indexes.",
+	Args:   cobra.NoArgs,
+	Hidden: true, // remove this once multi-index is enabled
+}
+
+var indexListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List configured indexes",
+	Long: `Print a list of configured indexes.
+
+This command prints a list of indexes. It shows the name and the remote URL for
+each configured index in table format.`,
+	Args: cobra.NoArgs,
+	RunE: func(_ *cobra.Command, _ []string) error {
+		dirs, err := ioutil.ReadDir(paths.IndexBase())
+		if err != nil {
+			return errors.Wrapf(err, "failed to read directory %s", paths.IndexBase())
+		}
+		var rows [][]string
+		for _, dir := range dirs {
+			indexName := dir.Name()
+			remote, err := gitutil.GetRemoteURL(paths.IndexPath(indexName))
+			if err != nil {
+				return errors.Wrapf(err, "failed to list the remote URL for index %s", indexName)
+			}
```

Comment:
Soon you'll find out you need a package where this lives and is tested.

I propose a package that does

```
ListIndexes(path string) ([]Index, error)
RemoveIndex(path, name string) error
AddIndex...
```

We can start small, but let's add tests.

Right now the code I just highlighted here has 0 tests.